### ISSUE TITLE
Automatically update order totals when editing an order (configurable)

### ISSUE THREAD

**Issue by mariannk** [nopSolutions/nopCommerce#8]
Automatically update order totals when editing an order in admin area (editing, deleting products, etc). 

But as we may have quite complex discount rules and tax rules which could be changed since an order was placed, then we should allow a store owner to choose whether order totals should be also updated (make this option configurable) 

It is really VERY complex to tackle it since it has to consider the products and their atributes and their inventory, weight, attributes and thier price/weight adjustments; regular, old and tiered prices, special prices and discount and tax rules (all which change from time to time); shipping expenses etc. etc. 
In my experience it is very uncommon. If you have inventory control (which is very important for certain kinds of products which have limited availavility, such as clothes, shoes, etc.) then changes are rare. The number of votes and requests for that are indicative that there is not much demand for it.

we should also update shipping info - http://www.nopcommerce.com/boards/t/43139/adding-products-to-existing-order-without-all-product-information.aspx

**Comment by AndreiMaz** [comment]
we should also update shipping info - http://www.nopcommerce.com/boards/t/43139/adding-products-to-existing-order-without-all-product-information.aspx

**Comment by johannwalder** [comment]
Hi,

how should this feature work? I understand it is still in beta but as it is closed I have tested it and noticed that it doesn't work when the product quantity is more than 1.

Therefore I am just wondering what should work and what is not supported.

Thanks!

BR,
Johann

**Comment by AndreiMaz** [comment]
@johannwalder what exactly doesn't work? please provide list of steps to reproduce the issue

**Comment by johannwalder** [comment]
@AndreiMaz thanks for the fast reply. Actually, it works fine after I have tested it again. My mistake was that I thought that the product quantity will automatically calculate the total of the product itself. When I manually add the correct price for the product totals then it also has the correct order totals.

Perhaps an idea for another work item in case it doesn't fit here. To allow adding/removing/editing products in the orders without the need to calculate the totals of the product.

Thanks again!

BR,
Johann

**Comment by pantonis** [comment]
I went to admin configuration order settings and I see an option there Auto update order totals which says that it is in beta mode. Is that setting still in beta mode?

**Comment by RomanovM** [comment]
Let's upgrade this feature from beta to viable version.

**Comment by nidhi1612** [comment]
Hi,
I am on nopcommerce 4.3 version. I understand it is in beta, but enabling the option doesn't even let me change the qty, without the option enabled I have to change all the totals manually, but enabling it does nothing. even qty remains same after changing it. Can you please help me with this until the stable feature comes out?

**Comment by brandon71** [comment]
This is a critical component for us as we have many back end users. A lot of manual calculation errors happen due to all the manual work when modifying orders totals.  Any chance this makes the 4.50 cut?

**Comment by theworkstim** [comment]
Just upgraded to 4.5 and this option still seems to do nothing? Our clients often edit order details after an order is this still on the horizon or has it been left.

**Comment by RomanovM** [comment]
See also: https://www.nopcommerce.com/boards/topic/94288/after-editing-order-checkout-attribute-disappears-from-total-count

**Comment by noptools** [comment]
**Need to call an inactive shipping rate caclulation plugin.**

Shipping Director requires that it be the only "Active" shipping rate caclulation plugin.  In Shipping Director's Option rule to call another shipping plugin (e.g., Shipping.UPS), SD preserves that external plugin system name (rather than using Shipping.Director), so that the "Package Tracking" feature of that plugin can be used.   The NopCommerce team resolved related issue here
[https://www.nopcommerce.com/en/boards/topic/37036/shippingservice-sets-the-systemname-please-change-to-do-that-conditionally#179512](url)

When editing an order the core Auto Update Order Total won't call an inactive shipping plugin and it fails.  Please allow the calling of an inactive plugin similar to change made above to allow tracking.

**Comment by rccwebmedia** [comment]
Yes, it appears that this feature does not allow or account for quantities. 
Even if I manually enter the including tax and excluding tax price while adding a product with a quantity of 2 the shipping is still using a quantity of one when using the basic Shipping by Total plugin. 
Also, when adding a product , if you increase the quantity to 2, the totals should auto-update on this screen so the correct total for that product gets entered instead of having to manually do the total.  
Lastly it is possible to add a product with a quantity of zero and it still adds the single unit price to the total.
For these reasons I do not think the quantity is being considered at all during the whole process so totals have to be done manually in end. Having to do the math manually defeats the whole purpose of the feature.

**Comment by brandon71** [comment]
What is the current status of this feature. Are there plans to support recalculation of totals in order management? It keeps getting bumped and is not currently on any milestone. Customers call in to make changes to orders. It just happens. It is very difficult for someone to recalculate everything by hand and get the order totals correct, especially with tier pricing.

**Comment by muhammedtas** [comment]
Hello Nop team. 
Any updates on this item? 
I need this update a lot. Thanks

**Comment by astrumcs** [comment]
One item I ran across today in testing the current implementation in 4.70.1 was that there is a good chance of losing information about the original payment values.  This has consequences for possible refunds and gift card balances.

For example, if you use a combination of a gift card and a check/money order, then remove an item, the order totals get updated, but you lose the knowledge of the initial payments.  So, if your initial order has two items, one for $40 and one for $20.  Then apply a $45 gift card for payment and then the remainder using a $15 check.  Later, you return the $40 item by deleting the item.  The check payment is zeroed out and then nothing happens with the gift card.  The consequence appears to be that the gift card balance does not get a $25 credit and you have no record of the check payment.

I bring this up as an awareness item without a proposed change or fix.  People just need to know about it.

**Comment by ksbhadani** [comment]
Also there is an issue with discounts applied using coupon codes, when updating order total those discounts gets removed
